src/Language/PureScript/CodeGen/JS.hs

```diff
@@ -414,5 +415,19 @@ moduleToJs (Module coms mn imps exps foreigns decls) foreign_ =
       js <- binderToJs elVar done'' binder
       return (JSVariableIntroduction elVar (Just (JSIndexer (JSNumericLiteral (Left index)) (JSVar varName))) : js)
 
+  -- Check that all integers fall within the valid int range for JavaScript.
+  checkIntegers :: JS -> m ()
+  checkIntegers = void . everywhereOnJSTopDownM go
+    where
+    go :: JS -> m JS
+    go (JSUnary Negate (JSNumericLiteral (Left i))) = return $ JSNumericLiteral (Left (-i))
```

Comment:
This is the only monadic traversal to hand for `JS`, and yeah this case is useful as otherwise the case below errors. Without moving the negation into the value we end up with `JSUnary Negate (JSNumericLiteral (Left 2147483648))`, which is larger than `maxInt`.